### PR TITLE
Master stock assign picking jip

### PR DESCRIPTION
**Pull request by jip-odoo**
Task: https://www.odoo.com/web?#id=1938107&action=327&model=project.task&view_type=form&menu_id=4720

Pad: https://pad.odoo.com/p/r.2cbe92f9461a2dc81de7095b70f86679 
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/wizard/stock_picking_responsible.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from odoo import api, fields, models
```
Comment (arbaes) [inline addons/stock/wizard/stock_picking_responsible.py]: Missing copyright

File: addons/stock/wizard/stock_picking_responsible.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class PickingResponsible(models.TransientModel):
```
Comment (arbaes) [inline addons/stock/wizard/stock_picking_responsible.py]: class name should be  `StockPickingResponsible`

File: addons/stock/wizard/stock_picking_responsible.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    _description = 'Assign Responsible'
+
+    user_id = fields.Many2one(
+        'res.users', 'Responsible',
+        domain=lambda self: [('groups_id', 'in', self.env.ref('stock.group_stock_user').id)],
+        default=lambda self: self.env.user)
+
+    @api.multi
+    def assign_responsible(self):
+        Pickings = self.env['stock.picking'].browse(self.env.context.get('active_ids'))
```
Comment (arbaes) [inline addons/stock/wizard/stock_picking_responsible.py]: pickings

File: addons/stock/wizard/stock_picking_responsible.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        for picking in Pickings:
```
Comment (arbaes) [inline addons/stock/wizard/stock_picking_responsible.py]: Use batch write 
```python3
picking.write({'user_id': self.user_id.id}) 
```

File: addons/stock/wizard/stock_picking_responsible.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+class PickingResponsible(models.TransientModel):
+    _name = 'stock.picking.responsible'
+    _description = 'Assign Responsible'
+
+    user_id = fields.Many2one(
+        'res.users', 'Responsible',
+        domain=lambda self: [('groups_id', 'in', self.env.ref('stock.group_stock_user').id)],
+        default=lambda self: self.env.user)
+
+    @api.multi
```
Comment (arbaes) [inline addons/stock/wizard/stock_picking_responsible.py]: Implicit decorator, remove

File: addons/stock/wizard/stock_picking_responsible.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    def assign_responsible(self):
```
Comment (arbaes) [inline addons/stock/wizard/stock_picking_responsible.py]: ensure_one

File: addons/stock/models/stock_picking.py [hunk 7]
```diff
@@ -265,6 +265,10 @@ class Picking(models.Model):
         default=lambda self: self.env['res.company']._company_default_get('stock.picking'),
         index=True, required=True,
         states={'done': [('readonly', True)], 'cancel': [('readonly', True)]})
+    user_id = fields.Many2one(
+        'res.users', 'Responsible', tracking=True,
+        domain=lambda self: [('groups_id', 'in', self.env.ref('stock.group_stock_user').id)],
+        default=lambda self: self.env.user)
```
Comment (arbaes) [inline addons/stock/models/stock_picking.py]: @jip-odoo what if the picking is created by a reording rule ? who will be the responsible ?

Comment (jip-odoo) [reply addons/stock/models/stock_picking.py]: i have updated changes as per your comment
if picking is create automatically in this case no one will be the responsible

**Comment by sle-odoo** [comment]
robodoo r+

**Comment by sle-odoo** [comment]
robodoo r+

**Pull request closed without merging**